source/server-discovery-and-monitoring/server-monitoring.rst

```diff
@@ -667,6 +667,12 @@ The event API here is assumed to be like the standard `Python Event
                 # Wait before running the next check.
                 wait()
                 continue
+
+            # for drivers that implement CMAP, mark the connection pool as ready
+            # once the server transitions to a known state.
```

Comment:
Ah good catch. Modifying connection pool state in `onServerDescriptionChanged` might be slightly problematic for some drivers though, since that is probably a method on the `TopologyDescription` instead of on the monitor. Would Python's implementation be able to plumb the connection pool through to its equivalent of that method? I think Rust could, but I'm not sure if that's true for every driver.

I'm wondering if marking the pool as ready here before calling that method might be okay, seeing as marking the pool as ready is correct so long as the server is _up_ (doesn't matter if it's a secondary / primary or if the status of another server changes). The only bad case I can think of is if between the server responding to the check successfully and that response actually getting back to the monitor, the server being checked went down and it was reported via another means. This seems rare enough to me that it might be okay to just mark the pool as ready without checking the topology version. What do you think?

Reply:
Updated the server monitoring spec to explicitly require this happening before `onServerDescriptionChanged`. I also updated the CMAP spec to more clearly describe the various behaviors of the pool.